From a37023aa028e9d39a3a6f76a8c332f310306c3ec Mon Sep 17 00:00:00 2001
From: Uwe Augustin <Info@FlapsOnline.de>
Date: Tue, 6 Oct 2026 17:50:15 +0200
Subject: [PATCH 16/22] OpenVario: remove OpenVarioMenu, and move src/OV to
 src/OpenVario

OpenVarioMenu is a start program for the OpenVario that offers a
handful of system functions next to "Start XCSoar".  Those functions
are part of XCSoar's own configuration now: the calibrations, the
firmware upgrade, the system backup and the choice of the program to
start on the Display and Services pages, the services on Services and
the data transfer in the Data Management dialog.  The
OpenVario image starts XCSoar from ovmenu-ng.sh; OpenVarioMenu is only
installed by the ovmenu-x package of the testing image, which has to
be dropped there together with this change.  Keeping a second program
with its own copy of these functions would only let the two drift
apart.

The files that XCSoar uses move from src/OV to src/OpenVario, the
name the rest of the code base uses for the device.  The functions
of OV/System that nothing calls any more (brightness, rotating the
display, the SSH switch) go with OpenVarioMenu, because the Display
and Services pages do all of that already.
---
 Makefile                                      |   1 -
 build/main.mk                                 |   4 +-
 build/ov.mk                                   |  47 --
 .../Settings/Panels/DisplayConfigPanel.cpp    |   4 +-
 .../Settings/Panels/OpenVarioSystemWidget.cpp |   4 +-
 .../Settings/Panels/SystemdConfigPanel.cpp    |   2 +-
 src/OV/OpenVarioMenu.cpp                      | 412 ------------------
 src/{OV => OpenVario}/Calibrate.cpp           |   0
 src/{OV => OpenVario}/Calibrate.hpp           |   0
 src/{OV => OpenVario}/System.cpp              |  96 ----
 src/{OV => OpenVario}/System.hpp              |  30 --
 test/src/TestOpenVarioImageName.cpp           |   2 +-
 12 files changed, 8 insertions(+), 594 deletions(-)
 delete mode 100644 build/ov.mk
 delete mode 100644 src/OV/OpenVarioMenu.cpp
 rename src/{OV => OpenVario}/Calibrate.cpp (100%)
 rename src/{OV => OpenVario}/Calibrate.hpp (100%)
 rename src/{OV => OpenVario}/System.cpp (56%)
 rename src/{OV => OpenVario}/System.hpp (87%)

diff --git a/Makefile b/Makefile
index 9d8075fe3..780eb9512 100644
--- a/Makefile
+++ b/Makefile
@@ -224,7 +224,6 @@ ifeq ($(TARGET_IS_LINUX),y)
 include $(topdir)/build/cloud.mk
 include $(topdir)/build/kobo.mk
 ifeq ($(USE_POLL_EVENT)$(TARGET_IS_KOBO),yn)
-include $(topdir)/build/ov.mk
 endif
 endif
 
diff --git a/build/main.mk b/build/main.mk
index d363eb00b..2128cb428 100644
--- a/build/main.mk
+++ b/build/main.mk
@@ -842,8 +842,8 @@ XCSOAR_SOURCES += \
 	$(SRC)/Linux/SystemdServiceList.cpp \
 	$(SRC)/Dialogs/Settings/Panels/OpenVarioSystemWidget.cpp \
 	$(SRC)/Dialogs/ProcessDialog.cpp \
-	$(SRC)/OV/System.cpp \
-	$(SRC)/OV/Calibrate.cpp
+	$(SRC)/OpenVario/System.cpp \
+	$(SRC)/OpenVario/Calibrate.cpp
 
 ifeq ($(USE_POLL_EVENT),y)
 SYSTEM_TOOLS_DIRECTORY ?= /usr/libexec/xcsoar/system-tools
diff --git a/build/ov.mk b/build/ov.mk
deleted file mode 100644
index 3d32fe8a1..000000000
--- a/build/ov.mk
+++ /dev/null
@@ -1,47 +0,0 @@
-OV_MENU_SOURCES = \
-	$(SRC)/Version.cpp \
-	$(SRC)/Asset.cpp \
-	$(SRC)/DataFilePath.cpp \
-	$(SRC)/LocalPath.cpp \
-	$(SRC)/Repository/FileType.cpp \
-	$(SRC)/FlightInfo.cpp \
-	$(SRC)/Formatter/HexColor.cpp \
-	$(SRC)/Formatter/TimeFormatter.cpp \
-	$(SRC)/Hardware/CPU.cpp \
-	$(SRC)/Hardware/DisplayDPI.cpp \
-	$(SRC)/Hardware/RotateDisplay.cpp \
-	$(SRC)/Screen/Layout.cpp \
-	$(SRC)/ui/control/TerminalWindow.cpp \
-	$(SRC)/Look/TerminalLook.cpp \
-	$(SRC)/Look/DialogLook.cpp \
-	$(SRC)/Look/ButtonLook.cpp \
-	$(SRC)/Look/CheckBoxLook.cpp \
-	$(SRC)/Renderer/TwoTextRowsRenderer.cpp \
-	$(SRC)/Renderer/FlightListRenderer.cpp \
-	$(SRC)/Logger/FlightParser.cpp \
-	$(SRC)/Gauge/LogoView.cpp \
-	$(SRC)/Dialogs/DialogSettings.cpp \
-	$(SRC)/Dialogs/WidgetDialog.cpp \
-	$(SRC)/Dialogs/HelpDialog.cpp \
-	$(SRC)/Dialogs/Message.cpp \
-	$(SRC)/Dialogs/LockScreen.cpp \
-	$(SRC)/Dialogs/TextEntry.cpp \
-	$(SRC)/Dialogs/KnobTextEntry.cpp \
-	$(SRC)/Dialogs/TouchTextEntry.cpp \
-	$(SRC)/Dialogs/ProcessDialog.cpp \
-	$(SRC)/Dialogs/Error.cpp \
-	$(TEST_SRC_DIR)/Fonts.cpp \
-	$(TEST_SRC_DIR)/FakeLanguage.cpp \
-	$(TEST_SRC_DIR)/FakeLogFile.cpp \
-	$(SRC)/Kobo/FakeSymbols.cpp \
-	$(SRC)/OV/System.cpp \
-	$(SRC)/OV/Calibrate.cpp \
-	$(SRC)/OV/OpenVarioMenu.cpp
-OV_MENU_DEPENDS = DBUS WIDGET FORM DATA_FIELD SCREEN EVENT RESOURCE ASYNC LIBNET OS IO THREAD TIME MATH UTIL
-OV_MENU_STRIP = y
-
-$(eval $(call link-program,OpenVarioMenu,OV_MENU))
-
-ifeq ($(TARGET),UNIX)
-OPTIONAL_OUTPUTS += $(OV_MENU_BIN)
-endif
diff --git a/src/Dialogs/Settings/Panels/DisplayConfigPanel.cpp b/src/Dialogs/Settings/Panels/DisplayConfigPanel.cpp
index edeb4ec48..c0c87266e 100644
--- a/src/Dialogs/Settings/Panels/DisplayConfigPanel.cpp
+++ b/src/Dialogs/Settings/Panels/DisplayConfigPanel.cpp
@@ -35,8 +35,8 @@
 #endif
 
 #if defined(__linux__) && !defined(__ANDROID__) && !defined(KOBO)
-#include "OV/Calibrate.hpp"
-#include "OV/System.hpp"
+#include "OpenVario/Calibrate.hpp"
+#include "OpenVario/System.hpp"
 #include "ui/event/Globals.hpp"
 #endif
 
diff --git a/src/Dialogs/Settings/Panels/OpenVarioSystemWidget.cpp b/src/Dialogs/Settings/Panels/OpenVarioSystemWidget.cpp
index fefc0c733..49b0d87e9 100644
--- a/src/Dialogs/Settings/Panels/OpenVarioSystemWidget.cpp
+++ b/src/Dialogs/Settings/Panels/OpenVarioSystemWidget.cpp
@@ -9,8 +9,8 @@
 #include "Form/Form.hpp"
 #include "Interface.hpp"
 #include "Language/Language.hpp"
-#include "OV/Calibrate.hpp"
-#include "OV/System.hpp"
+#include "OpenVario/Calibrate.hpp"
+#include "OpenVario/System.hpp"
 #include "Profile/Profile.hpp"
 #include "Profile/ProfileMap.hpp"
 #include "UIGlobals.hpp"
diff --git a/src/Dialogs/Settings/Panels/SystemdConfigPanel.cpp b/src/Dialogs/Settings/Panels/SystemdConfigPanel.cpp
index 121242dd6..eaf9cf583 100644
--- a/src/Dialogs/Settings/Panels/SystemdConfigPanel.cpp
+++ b/src/Dialogs/Settings/Panels/SystemdConfigPanel.cpp
@@ -3,7 +3,7 @@
 
 #include "SystemdConfigPanel.hpp"
 #include "OpenVarioSystemWidget.hpp"
-#include "OV/System.hpp"
+#include "OpenVario/System.hpp"
 #include "Widget/TwoWidgets.hpp"
 #include "Dialogs/Error.hpp"
 #include "Dialogs/JobDialog.hpp"
diff --git a/src/OV/OpenVarioMenu.cpp b/src/OV/OpenVarioMenu.cpp
deleted file mode 100644
index 2a0e8033a..000000000
--- a/src/OV/OpenVarioMenu.cpp
+++ /dev/null
@@ -1,412 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
-// Copyright The XCSoar Project
-
-#include "Dialogs/DialogSettings.hpp"
-#include "Dialogs/Message.hpp"
-#include "Dialogs/WidgetDialog.hpp"
-#include "Dialogs/ProcessDialog.hpp"
-#include "Dialogs/Error.hpp"
-#include "Widget/DrawWidget.hpp"
-#include "Widget/RowFormWidget.hpp"
-#include "Renderer/FlightListRenderer.hpp"
-#include "UIGlobals.hpp"
-#include "Look/DialogLook.hpp"
-#include "Screen/Layout.hpp"
-#include "../test/src/Fonts.hpp"
-#include "ui/canvas/Canvas.hpp"
-#include "ui/window/Init.hpp"
-#include "ui/window/SingleWindow.hpp"
-#include "ui/event/Queue.hpp"
-#include "ui/event/Timer.hpp"
-#include "Logger/FlightParser.hpp"
-#include "Language/Language.hpp"
-#include "DataFilePath.hpp"
-#include "system/Process.hpp"
-#include "io/FileLineReader.hpp"
-#include "util/PrintException.hxx"
-#include "util/ScopeExit.hxx"
-#include "LocalPath.hpp"
-#include "OV/Calibrate.hpp"
-
-#include <cassert>
-#include <cstdio>
-
-enum Buttons {
-  LAUNCH_SHELL = 100,
-};
-
-static bool have_data_path = false;
-
-static DialogSettings dialog_settings;
-static UI::SingleWindow *global_main_window;
-static DialogLook *global_dialog_look;
-
-const DialogSettings &
-UIGlobals::GetDialogSettings()
-{
-  return dialog_settings;
-}
-
-const DialogLook &
-UIGlobals::GetDialogLook()
-{
-  assert(global_dialog_look != nullptr);
-
-  return *global_dialog_look;
-}
-
-UI::SingleWindow &
-UIGlobals::GetMainWindow()
-{
-  assert(global_main_window != nullptr);
-
-  return *global_main_window;
-}
-
-class FileMenuWidget final
-  : public RowFormWidget
-{
-  UI::Display &display;
-  UI::EventQueue &event_queue;
-
-public:
-  FileMenuWidget(UI::Display &_display, UI::EventQueue &_event_queue,
-                 const DialogLook &look) noexcept
-    :RowFormWidget(look),
-     display(_display), event_queue(_event_queue) {}
-
-private:
-  /* virtual methods from class Widget */
-  void Prepare(ContainerWindow &parent,
-               const PixelRect &rc) noexcept override;
-};
-
-void
-FileMenuWidget::Prepare([[maybe_unused]] ContainerWindow &parent,
-                        [[maybe_unused]] const PixelRect &rc) noexcept
-{
-  AddButton("Download XCSoar IGC files to USB", [this](){
-    const UI::ScopeDropMaster drop_master{display};
-    const UI::ScopeSuspendEventQueue suspend_event_queue{event_queue};
-    Run("/usr/bin/download-igc.sh");
-  });
-
-  AddButton("Download XCSoar to USB", [](){
-    static constexpr const char *argv[] = {
-      "/usr/bin/download-all.sh", nullptr
-    };
-
-    RunProcessDialog(UIGlobals::GetMainWindow(),
-                     UIGlobals::GetDialogLook(),
-                     "Downloading files", argv);
-  });
-
-  AddButton("Upload files from USB to XCSoar", [](){
-    static constexpr const char *argv[] = {
-      "/usr/bin/upload-xcsoar.sh", nullptr
-    };
-
-    RunProcessDialog(UIGlobals::GetMainWindow(),
-                     UIGlobals::GetDialogLook(),
-                     "Uploading files", argv);
-  });
-}
-
-class SystemMenuWidget final
-  : public RowFormWidget
-{
-  UI::Display &display;
-  UI::EventQueue &event_queue;
-
-public:
-  SystemMenuWidget(UI::Display &_display, UI::EventQueue &_event_queue,
-                   const DialogLook &look) noexcept
-    :RowFormWidget(look),
-     display(_display), event_queue(_event_queue) {}
-
-private:
-  /* virtual methods from class Widget */
-  void Prepare(ContainerWindow &parent,
-               const PixelRect &rc) noexcept override;
-};
-
-void
-SystemMenuWidget::Prepare([[maybe_unused]] ContainerWindow &parent,
-                          [[maybe_unused]] const PixelRect &rc) noexcept
-{
-  AddButton("Update System", [](){
-    static constexpr const char *argv[] = {
-      "/usr/bin/update-system.sh", nullptr
-    };
-
-    RunProcessDialog(UIGlobals::GetMainWindow(),
-                     UIGlobals::GetDialogLook(),
-                     "Update System", argv);
-  });
-
-  AddButton("Update Maps", [](){
-    static constexpr const char *argv[] = {
-      "/usr/bin/update-maps.sh", nullptr
-    };
-
-    RunProcessDialog(UIGlobals::GetMainWindow(),
-                     UIGlobals::GetDialogLook(),
-                     "Update Maps", argv);
-  });
-
-  AddButton("Calibrate Sensors", CalibrateSensors);
-  AddButton("Calibrate Touch", [this](){
-    CalibrateTouch(display, event_queue);
-  });
-
-  AddButton("System Settings", [this](){
-    const UI::ScopeDropMaster drop_master{display};
-    const UI::ScopeSuspendEventQueue suspend_event_queue{event_queue};
-    Run("/usr/lib/openvario/libexec/system_settings.sh");
-  });
-
-  AddButton("System Info", [this](){
-    const UI::ScopeDropMaster drop_master{display};
-    const UI::ScopeSuspendEventQueue suspend_event_queue{event_queue};
-    Run("/usr/lib/openvario/libexec/system_info.sh");
-  });
-}
-
-class MainMenuWidget final
-  : public RowFormWidget
-{
-  enum Controls {
-    XCSOAR,
-    LOGBOOK,
-    FILE,
-    SYSTEM,
-    SHELL,
-    REBOOT,
-    SHUTDOWN,
-    TIMER,
-  };
-
-  UI::Display &display;
-  UI::EventQueue &event_queue;
-
-  WndForm &dialog;
-
-  UI::Timer timer{[this](){
-    if (--remaining_seconds == 0) {
-      HideRow(Controls::TIMER);
-      StartXCSoar();
-    } else {
-      ScheduleTimer();
-    }
-  }};
-
-  unsigned remaining_seconds = 3;
-
-public:
-  MainMenuWidget(UI::Display &_display, UI::EventQueue &_event_queue,
-                 WndForm &_dialog) noexcept
-    :RowFormWidget(_dialog.GetLook()),
-     display(_display), event_queue(_event_queue),
-     dialog(_dialog) {}
-
-private:
-  void StartXCSoar() noexcept {
-    const UI::ScopeDropMaster drop_master{display};
-    const UI::ScopeSuspendEventQueue suspend_event_queue{event_queue};
-    Run("/usr/bin/xcsoar", "-fly");
-  }
-
-  void ScheduleTimer() noexcept {
-    assert(remaining_seconds > 0);
-
-    timer.Schedule(std::chrono::seconds{1});
-
-    char buffer[256];
-    snprintf(buffer, sizeof(buffer), "Starting XCSoar in %u seconds (press any key to cancel)",
-             remaining_seconds);
-    SetText(Controls::TIMER, buffer);
-  }
-
-  void CancelTimer() noexcept {
-    timer.Cancel();
-    remaining_seconds = 0;
-    HideRow(Controls::TIMER);
-  }
-
-  /* virtual methods from class Widget */
-  void Prepare(ContainerWindow &parent,
-               const PixelRect &rc) noexcept override;
-
-  void Show(const PixelRect &rc) noexcept override {
-    RowFormWidget::Show(rc);
-
-    if (remaining_seconds > 0)
-      ScheduleTimer();
-  }
-
-  void Hide() noexcept override {
-    CancelTimer();
-    RowFormWidget::Hide();
-  }
-
-  bool KeyPress(unsigned key_code) noexcept override {
-    CancelTimer();
-    return RowFormWidget::KeyPress(key_code);
-  }
-};
-
-static void
-ShowLogbook() noexcept
-{
-  const auto &look = UIGlobals::GetDialogLook();
-  FlightListRenderer renderer{look.text_font, look.bold_font};
-
-  try {
-    FileLineReaderA file(ResolveLogsDataPath("flights.log"));
-
-    FlightParser parser{file};
-    FlightInfo flight;
-    while (parser.Read(flight))
-      renderer.AddFlight(flight);
-  } catch (...) {
-    ShowError(std::current_exception(), "Logbook");
-    return;
-  }
-
-  TWidgetDialog<DrawWidget>
-    sub_dialog(WidgetDialog::Full{}, UIGlobals::GetMainWindow(),
-               look, "Logbook");
-
-  sub_dialog.SetWidget([&renderer](Canvas &canvas, const PixelRect &rc){
-    renderer.Draw(canvas, rc);
-  });
-
-  sub_dialog.AddButton(_("Close"), mrOK);
-  sub_dialog.ShowModal();
-}
-
-void
-MainMenuWidget::Prepare([[maybe_unused]] ContainerWindow &parent,
-			[[maybe_unused]] const PixelRect &rc) noexcept
-{
-  AddButton("Start XCSoar", [this](){
-    CancelTimer();
-    StartXCSoar();
-  });
-
-  if (have_data_path)
-      AddButton("Logbook", [this](){
-        CancelTimer();
-        ShowLogbook();
-      });
-  else
-    AddDummy();
-
-  AddButton("Files", [this](){
-    CancelTimer();
-
-    TWidgetDialog<FileMenuWidget>
-      sub_dialog(WidgetDialog::Full{}, dialog.GetMainWindow(),
-                 GetLook(), "OpenVario Files");
-    sub_dialog.SetWidget(display, event_queue, GetLook());
-    sub_dialog.AddButton(_("Close"), mrOK);
-    return sub_dialog.ShowModal();
-  });
-
-  AddButton("System", [this](){
-    CancelTimer();
-
-    TWidgetDialog<SystemMenuWidget>
-      sub_dialog(WidgetDialog::Full{}, dialog.GetMainWindow(),
-                 GetLook(), "OpenVario System");
-    sub_dialog.SetWidget(display, event_queue, GetLook());
-    sub_dialog.AddButton(_("Close"), mrOK);
-    return sub_dialog.ShowModal();
-  });
-
-  AddButton("Shell", [this](){
-    dialog.SetModalResult(LAUNCH_SHELL);
-  });
-
-  AddButton("Reboot", [](){
-    Run("/sbin/reboot");
-  });
-
-  AddButton("Power off", [](){
-    Run("/sbin/poweroff");
-  });
-
-  AddReadOnly("");
-}
-
-static int
-Main(UI::EventQueue &event_queue, UI::SingleWindow &main_window,
-     const DialogLook &dialog_look)
-{
-  TWidgetDialog<MainMenuWidget>
-    dialog(WidgetDialog::Full{}, main_window,
-           dialog_look, "OpenVario");
-  dialog.SetWidget(main_window.GetDisplay(), event_queue, dialog);
-
-  return dialog.ShowModal();
-}
-
-static int
-Main()
-{
-  dialog_settings.SetDefaults();
-
-  ScreenGlobalInit screen_init;
-  Layout::Initialise(screen_init.GetDisplay(), {600, 800});
-  InitialiseFonts();
-
-  DialogLook dialog_look;
-  dialog_look.Initialise();
-
-  UI::TopWindowStyle main_style;
-  main_style.Resizable();
-
-  UI::SingleWindow main_window{screen_init.GetDisplay()};
-  main_window.Create("XCSoar/OpenVarioMenu", {600, 800}, main_style);
-  main_window.Show();
-
-  global_dialog_look = &dialog_look;
-  global_main_window = &main_window;
-
-  int action = Main(screen_init.GetEventQueue(), main_window, dialog_look);
-
-  main_window.Destroy();
-
-  DeinitialiseFonts();
-
-  return action;
-}
-
-int main()
-{
-  try {
-    InitialiseDataPath();
-    have_data_path = true;
-  } catch (...) {
-    fprintf(stderr, "Failed to locate data path: ");
-    PrintException(std::current_exception());
-  }
-
-  AtScopeExit() {
-    if (have_data_path)
-      DeinitialiseDataPath();
-  };
-
-  int action = Main();
-
-  switch (action) {
-  case LAUNCH_SHELL:
-    execl("/bin/bash", "bash", "--login", nullptr);
-    execl("/bin/ash", "-ash", nullptr);
-    execl("/bin/ash", "-sh", nullptr);
-    perror("Failed to launch shell");
-    return EXIT_FAILURE;
-  }
-
-  return action;
-}
diff --git a/src/OV/Calibrate.cpp b/src/OpenVario/Calibrate.cpp
similarity index 100%
rename from src/OV/Calibrate.cpp
rename to src/OpenVario/Calibrate.cpp
diff --git a/src/OV/Calibrate.hpp b/src/OpenVario/Calibrate.hpp
similarity index 100%
rename from src/OV/Calibrate.hpp
rename to src/OpenVario/Calibrate.hpp
diff --git a/src/OV/System.cpp b/src/OpenVario/System.cpp
similarity index 56%
rename from src/OV/System.cpp
rename to src/OpenVario/System.cpp
index 1425fc4cc..6e14dbc3f 100644
--- a/src/OV/System.cpp
+++ b/src/OpenVario/System.cpp
@@ -2,21 +2,14 @@
 // Copyright The XCSoar Project
 
 #include "System.hpp"
-#include "lib/dbus/Connection.hxx"
-#include "lib/dbus/ScopeMatch.hxx"
-#include "lib/dbus/Systemd.hxx"
 #include "system/FileUtil.hpp"
 #include "system/Path.hpp"
 #include "io/KeyValueFileReader.hpp"
 #include "io/FileOutputStream.hxx"
 #include "io/BufferedOutputStream.hxx"
 #include "io/FileLineReader.hpp"
-#include "Dialogs/Error.hpp"
 #include "DisplayOrientation.hpp"
-#include "Hardware/RotateDisplay.hpp"
 
-#include <unistd.h>
-#include <sys/stat.h>
 #include <fmt/format.h>
 
 #include <map>
@@ -52,46 +45,6 @@ WriteConfigFile(std::map<std::string, std::string, std::less<>> &map, Path path)
   file.Commit();
 }
 
-uint_least8_t
-OpenvarioGetBrightness() noexcept
-{
-  char line[4];
-  int result = 10;
-
-  if (File::ReadString(Path("/sys/class/backlight/lcd/brightness"), line, sizeof(line))) {
-    result = atoi(line);
-  }
-
-  return result;
-}
-
-void
-OpenvarioSetBrightness(uint_least8_t value) noexcept
-{
-  if (value < 1) { value = 1; }
-  if (value > 10) { value = 10; }
-
-  File::WriteExisting(Path("/sys/class/backlight/lcd/brightness"), fmt::format_int{value}.c_str());
-}
-
-DisplayOrientation
-OpenvarioGetRotation()
-{
-  std::map<std::string, std::string, std::less<>> map;
-  LoadConfigFile(map, Path(boot_config));
-
-  uint_least8_t result;
-  result = map.contains("rotation") ? std::stoi(map.find("rotation")->second) : 0;
-
-  switch (result) {
-  case 0: return DisplayOrientation::LANDSCAPE;
-  case 1: return DisplayOrientation::REVERSE_PORTRAIT;
-  case 2: return DisplayOrientation::REVERSE_LANDSCAPE;
-  case 3: return DisplayOrientation::PORTRAIT;
-  default: return DisplayOrientation::DEFAULT;
-  }
-}
-
 /**
  * The value of "rotation" in /boot/config.uEnv, which is also the
  * value of /sys/class/graphics/fbcon/rotate.
@@ -124,55 +77,6 @@ OpenvarioSaveRotation(DisplayOrientation orientation)
   WriteConfigFile(map, Path(boot_config));
 }
 
-void
-OpenvarioSetRotation(DisplayOrientation orientation)
-{
-  Display::Rotate(orientation);
-
-  File::WriteExisting(Path("/sys/class/graphics/fbcon/rotate"),
-                      fmt::format_int{ToConfigRotation(orientation)}.c_str());
-
-  OpenvarioSaveRotation(orientation);
-}
-
-SSHStatus
-OpenvarioGetSSHStatus()
-{
-  auto connection = ODBus::Connection::GetSystem();
-
-  if (Systemd::IsUnitEnabled(connection, "dropbear.socket")) {
-    return SSHStatus::ENABLED;
-  } else if (Systemd::IsUnitActive(connection, "dropbear.socket")) {
-    return SSHStatus::TEMPORARY;
-  } else {
-    return SSHStatus::DISABLED;
-  }
-}
-
-void
-OpenvarioEnableSSH(bool temporary)
-{
-  auto connection = ODBus::Connection::GetSystem();
-  const ODBus::ScopeMatch job_removed_match{connection, Systemd::job_removed_match};
-
-  if (temporary)
-    Systemd::DisableUnitFile(connection, "dropbear.socket");
-  else
-    Systemd::EnableUnitFile(connection, "dropbear.socket");
-
-  Systemd::StartUnit(connection, "dropbear.socket");
-}
-
-void
-OpenvarioDisableSSH()
-{
-  auto connection = ODBus::Connection::GetSystem();
-  const ODBus::ScopeMatch job_removed_match{connection, Systemd::job_removed_match};
-
-  Systemd::DisableUnitFile(connection, "dropbear.socket");
-  Systemd::StopUnit(connection, "dropbear.socket");
-}
-
 std::string
 OpenvarioGetMainApp()
 {
diff --git a/src/OV/System.hpp b/src/OpenVario/System.hpp
similarity index 87%
rename from src/OV/System.hpp
rename to src/OpenVario/System.hpp
index 79feba6a3..ec3a00c19 100644
--- a/src/OV/System.hpp
+++ b/src/OpenVario/System.hpp
@@ -11,11 +11,6 @@
 
 class Path;
 
-enum class SSHStatus {
-  ENABLED,
-  DISABLED,
-  TEMPORARY,
-};
 /**
  * Is this program running on an OpenVario?  The OpenVario image keeps
  * its boot settings (display rotation, the program to start) in
@@ -36,15 +31,6 @@ LoadConfigFile(std::map<std::string, std::string, std::less<>> &map, Path path);
 void
 WriteConfigFile(std::map<std::string, std::string, std::less<>> &map, Path path);
 
-uint_least8_t
-OpenvarioGetBrightness() noexcept;
-
-void
-OpenvarioSetBrightness(uint_least8_t value) noexcept;
-
-DisplayOrientation
-OpenvarioGetRotation();
-
 /**
  * Store the display orientation in /boot/config.uEnv, from where the
  * image applies it to the console at the next boot.
@@ -52,22 +38,6 @@ OpenvarioGetRotation();
 void
 OpenvarioSaveRotation(DisplayOrientation orientation);
 
-/**
- * Rotate the display and the console now, and store the orientation
- * for the next boot.
- */
-void
-OpenvarioSetRotation(DisplayOrientation orientation);
-
-SSHStatus
-OpenvarioGetSSHStatus();
-
-void
-OpenvarioEnableSSH(bool temporary);
-
-void
-OpenvarioDisableSSH();
-
 /**
  * The program the OpenVario image starts after boot: the value of
  * "main_app" in /boot/config.uEnv, for example "xcsoar" or
diff --git a/test/src/TestOpenVarioImageName.cpp b/test/src/TestOpenVarioImageName.cpp
index c94d38aef..1783b260d 100644
--- a/test/src/TestOpenVarioImageName.cpp
+++ b/test/src/TestOpenVarioImageName.cpp
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 // Copyright The XCSoar Project
 
-#include "OV/System.hpp"
+#include "OpenVario/System.hpp"
 #include "TestUtil.hpp"
 
 using std::string_view_literals::operator""sv;
-- 
2.43.0

